Library/Homebrew/test/download_strategies/curl_spec.rb

```diff
@@ -147,22 +125,48 @@
   end
 
   describe "#cached_location" do
+    subject(:cached_location) { described_class.new(url, name, version, **specs).cached_location }
+
+    context "when URL ends with file" do
+      it {
+        expect(cached_location).to eq(
+          HOMEBREW_CACHE/"downloads/3d1c0ae7da22be9d83fb1eb774df96b7c4da71d3cf07e1cb28555cf9a5e5af70--foo.tar.gz",
+        )
+      }
+    end
+
+    context "when URL file is in middle" do
+      let(:url) { "https://example.com/foo.tar.gz/from/this/mirror" }
+
+      it {
+        expect(cached_location).to eq(
+          HOMEBREW_CACHE/"downloads/1ab61269ba52c83994510b1e28dd04167a2f2e8393a35a9c50c1f7d33fd8f619--foo.tar.gz",
+        )
+      }
+    end
+
     context "with a file name trailing the URL path" do
       let(:url) { "https://example.com/cask.dmg" }
 
-      its("cached_location.extname") { is_expected.to eq(".dmg") }
+      it {
```

Comment:
```suggestion
      it "returns a file with a `.dmg` extension" {
```

Might be a good idea to add some sort of description like this to the `its` to `it` conversions.

Without a description:
```
CurlDownloadStrategy#cached_location with a file name trailing the URL path is expected to eq ".dmg"
```

With a description:
```
CurlDownloadStrategy#cached_location with a file name trailing the URL path returns a file with a `.dmg` extension
```

Reply:
I mostly didn't add descriptions as the other `it {`s that I moved down didn't have any. 🤷🏻